src/main/java/frc/robot/commands/LevelRobot.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.commands;
+
+import com.ctre.phoenix.sensors.WPI_Pigeon2;
+
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.ExampleSubsystem;
+import frc.robot.subsystems.Swerve.DrivebaseSubsystem;
+public class LevelRobot extends CommandBase{
+    private final DrivebaseSubsystem drive;
+    private final PIDController pid = new PIDController(
+        Constants.LevelRobotConstants.PID_kP, 
+        Constants.LevelRobotConstants.PID_kI, 
+        Constants.LevelRobotConstants.PID_kD);
+    private final PIDController headingPid = new PIDController(
+        Constants.LevelRobotConstants.HEADING_PID_kP,
+        Constants.LevelRobotConstants.HEADING_PID_kI,
+        Constants.LevelRobotConstants.HEADING_PID_kD);
+    private double startingYaw;
+    
+    public LevelRobot(SwerveSubsystem drivetrain){
+        drive = drivetrain;
+        addRequirements(drive);
+    }
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+        pid.setTolerance(0);
+        pid.setSetpoint(0);
+        startingYaw = drive.getYawAsDouble() % 360;
+    }
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+        double roll = drive.getRoll();
+        double yaw = drive.getYawAsDouble() % 360;
+        double deltaYaw = yaw-startingYaw;
```

Comment:
I feel here, PIDing `deltaYaw` to 0 is the same thing as PIDing directly to the setpoint. This is will work completely fine, but I think it might be a bit more complicated to read. But I am not fully sure if the behavior will be the same, might be worth testing?

Reply:
We used to run with PIDing to 0, but that doesnt work how we want as we want to allow for the the drive to drive on the opposite way. PIDing the yaw this way allows for the driver to have as mush freedom as possible without the PID fighting them.